### PR TITLE
build: use debian instead of apline to build on glibc instead of muslc

### PR DESCRIPTION
**Pull request by malnick**
Thank you for sending a pull request :heart:

### Checklist

Please review the [guidelines for contributing code](../CONTRIBUTING.md#contribute-code) to this repository. And check `[x]` all the boxes that apply:

- [ ] This request **pulls a feature/bugfix branch** (right side) from my fork. *Not my master.*
- [x] This request **pulls against** `ockam-network/ockam`’s **develop branch** (left side).
- [x] Build succeeds `./build` with no linter warnings or test failures.
- [x] All code matched our code formatting [conventions](../CONTRIBUTING.md#code-format).
- [x] Commit messages match our [conventions](../CONTRIBUTING.md#commit-messages) and all commits
are [signed](#signed-commits).
- [ ] This request includes tests and documentation for all new code.

### Proposed Changes
Please describe the changes in your pull request.

If this pull request resolves an already recorded bug or a feature request, be sure to link to that issue.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -44,9 +44,16 @@ CMD ["-a", "build"]
...
 # docker run --rm --volume "$(pwd):/project" ockam/tool/go:latest
-FROM golang:1.11.2-alpine3.8@sha256:692eff58ac23cafc7cb099793feb00406146d187cd3ba0226809317952a9cf37 as go
+FROM golang:1.11.2-alpine3.8@sha256:692eff58ac23cafc7cb099793feb00406146d187cd3ba0226809317952a9cf37 as go-alpine
 ENV GOOS=linux GOARCH=amd64 CGO_ENABLED=1
-RUN apk --update add --no-cache gcc g++ libtool git
+RUN apk --update add --no-cache gcc g++ libtool git libgcc
+WORKDIR /project
+ENTRYPOINT ["go"]
+
+FROM golang:1.11.5-stretch as go
```
Comment (mrinalwadhwa) [inline Dockerfile]: pin the hash and let's stick with 1.11.2 for stretch as well ... we bump all images to lates go later.

File: Dockerfile [hunk 2]
```diff
@@ -44,9 +44,16 @@ CMD ["-a", "build"]
...
+ENV GOOS=linux GOARCH=amd64 CGO_ENABLED=1
+RUN apt-get update
+RUN apt-get install gcc g++ libtool git
```
Comment (mrinalwadhwa) [inline Dockerfile]: this line needs a `apt-get install -y`, the missing `-y` makes the build fail

Comment (mrinalwadhwa) [reply Dockerfile]: https://dev.azure.com/ockam-network/ockam/_build/results?buildId=416&view=logs&jobId=12f1170f-54f2-53f3-20dd-22fc7dff55f9&taskId=9c939e41-62c2-5605-5e05-fc3554afc9f5&lineStart=699&lineEnd=700&colStart=1&colEnd=1

**Comment by malnick** [comment]
@mrinalwadhwa This build now passes local `./build` and I verified we're linking with `ld-linux-x86-64.so.2` instead of muslc:

```
➜  ockam malnick/use-debian ✓ ./build
Building .build/ockam_0.3.0_linux_amd64 ... Done.
Building .build/ockam_0.3.0_linux_arm64 ... Done.
Building .build/ockam_0.3.0_darwin_amd64 ... Done.
Building .build/ockam_0.3.0_windows_amd64.exe ... Done.
➜  ockam malnick/use-debian ✓ file .build/ockam_0.3.0_linux_amd64 
.build/ockam_0.3.0_linux_amd64: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, stripped
```

**Comment by malnick** [comment]
Here's the output from the shared library deps:
```
➜  ockam malnick/use-debian ✓ ldd .build/ockam_0.3.0_linux_amd64 
        linux-vdso.so.1 (0x00007ffc9d52d000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fead13a6000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fead0fb5000)
        /lib64/ld-linux-x86-64.so.2 (0x00007fead15c5000)
```

Since I have `musl-dev` installed we'll probably want a sanity check this and execute it on a linux system that doesn't, though I'm confident with the above tests that we're a :+1: now.

File: Dockerfile [hunk 3]
```diff
@@ -76,7 +83,7 @@ CMD ["--vendor", "--enable-all", "--line-length=120", "./..."]
 #
 # DOCKER_BUILDKIT=1 docker build --target goreleaser --tag ockam/tool/goreleaser:latest .
 # docker run --rm --volume "$(pwd):/project" ockam/tool/goreleaser:latest
-FROM go as goreleaser
+FROM go-alpine as goreleaser
```
Comment (mrinalwadhwa) [inline Dockerfile]: so goreleaser builds the final binaries this likely needs to be the `go` debian image to fix the release binaries.

File: Dockerfile [hunk 4]
```diff
@@ -44,9 +44,16 @@ CMD ["-a", "build"]
 #
 # DOCKER_BUILDKIT=1 docker build --target go --tag ockam/tool/go:latest .
 # docker run --rm --volume "$(pwd):/project" ockam/tool/go:latest
-FROM golang:1.11.2-alpine3.8@sha256:692eff58ac23cafc7cb099793feb00406146d187cd3ba0226809317952a9cf37 as go
+FROM golang:1.11.2-alpine3.8@sha256:692eff58ac23cafc7cb099793feb00406146d187cd3ba0226809317952a9cf37 as go-alpine
 ENV GOOS=linux GOARCH=amd64 CGO_ENABLED=1
-RUN apk --update add --no-cache gcc g++ libtool git
+RUN apk --update add --no-cache gcc g++ libtool git libgcc
```
Comment (mrinalwadhwa) [inline Dockerfile]: remove libgcc to keep the alpine image the way it was before this change.

**Comment by malnick** [comment]
@mrinalwadhwa updated goreleaser to use debian and removed glibc from the alpine go image.

**Comment by mrinalwadhwa** [comment]
merged

**Pull request closed without merging**